### ISSUE TITLE
RequireFinalReturn, no return after if statement

### ISSUE THREAD

**Issue by benrubson** [Perl-Critic/Perl-Critic#783]
Hi,

```
$ perlcritic --version
1.130
```

Why `RequireFinalReturn` does not complain about the following ?
```
sub test
{
    if(0)
    {
        return;
    }
}
```

I may certainly be missing something, but there is a return statement inside the condition only.
So I would have thought we would require a return statement after the condition.

Thank you for your support and explanation 👍 

Ben

Edit : [RequireFinalReturn man page](https://metacpan.org/pod/Perl::Critic::Policy::Subroutines::RequireFinalReturn)

**Comment by benrubson** [comment]
Any thoughts ?
Thank you again 👍

**Comment by petdance** [comment]
> Why RequireFinalReturn does not complain about the following ?

**Comment by petdance** [comment]
I don't know why it doesn't complain about that construct.  I'd expect it to.  As far as I'm concerned, it's a bug.

**Comment by benrubson** [comment]
Thank you @petdance for your feedback, glad to see this may be corrected in a future release 👍

**Comment by benrubson** [comment]
Hi @petdance, some news about this perhaps ?
Many thanks 👍

**Comment by petdance** [comment]
No, I don't have any.  Someone needs to look into it and figure out what the problem is.

**Comment by petdance** [comment]
I'd like to leave this open, because it is indeed a problem, even though nobody has had time to address it.
